### PR TITLE
feat(pagination): migrate to TypeScript

### PR DESCRIPTION
**Pull request by austin94**
- [x] **BREAKING CHANGE**

## Description

This PR moves `react-pagination` to TypeScript. This component was highly-abstracted before this transition so there are few API changes.

### Breaking Changes

- All `View` components are not longer exported
- Removed the following components:
  - `Gap`
  - `NextPage`
  - `Page`
  - `PaginationView`
  - `PreviousPage`

## Checklist

- [ ] :ok_hand: design updates are Garden Designer approved (add the
      designer as a reviewer)
- [ ] :nail_care: view component styling is based on a Garden CSS
      component
- [ ] :globe_with_meridians: Styleguidist demo is up-to-date (`yarn start`)
- [ ] :arrow_left: renders as expected with reversed (RTL) direction
- [ ] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [ ] :guardsman: includes new unit tests
- [ ] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

**Comment by austin94** [comment]
https://austin-next.netlify.com/pagination/ is updated to show the new example layouts

**Comment by austin94** [comment]
@jzempel since you're queued up for the CSS-in-JS work I haven't modified the default examples yet in case you had some ideas there.

File: BREAKING_CHANGES.md [hunk 1]
```diff
@@ -125,6 +125,16 @@ New features/fixes include:
   - Use `duration` instead which accepts `ms` defaults to 1250ms
 - `<Skeleton />` loader has renamed `dark` prop to `isDark`
 
+## @zendeskgarden/react-pagination
+
+- All `View` components are not longer exported
```
Comment (jzempel) [inline BREAKING_CHANGES.md]: [nit] *no

Comment (austin94) [reply BREAKING_CHANGES.md]: Updated.

**Pull request merged**